Estore.UITests/StepDefinitions/UserSteps.cs

```diff
@@ -40,5 +50,93 @@ public void WhenUserSearchProductBy(string searchedString)
             }
             _context.MainPage.ClickSearchButton();
         }
+
+        [Given(@"User opens login page clicking on the Login button")]
+        public void GivenUserOpensLoginPageClickingOnTheLoginButton()
+        {
+            _context.CurrentPage.ClickLoginLink();                        
+        }
+
+        [Given(@"User fills email and password fields with '([^']*)' credentials")]
+        [When(@"User fills email and password fields with '([^']*)' credentials")]
+        public async Task WhenUserFillsEmailAndPasswordFieldsWithCredentials(string userRole)
+        {
+            switch (userRole)
+            {
+                case "Customer":
+                    _context.CurrentUser = await _userProvider.RegisterCustomer();
+                    _context.RegisteredCustomers.Add(_context.CurrentUser.Id.Value);
+                    _context.LoginPage.FillEmailAndPasswordFields(_context.CurrentUser.Credentials.Email,
+                        _context.CurrentUser.Credentials.Password);
+                    break;
+                case "Admin":
+                    _context.CurrentUser = _credentials.GetAdminCredentials();
+                    _context.LoginPage.FillEmailAndPasswordFields(_context.CurrentUser.Credentials.Email,
+                        _context.CurrentUser.Credentials.Password);
+                    break;
+                case "Empty":
+                    _context.LoginPage.FillEmailAndPasswordFields("", "");
+                    break;
+            }
+        }
+
+        [Given(@"User clicks Login button")]
+        [When(@"User clicks Login button")]
+        public void WhenUserClicksLoginButton()
```

Comment:
it's better to use name of method without "when" prefix. May be it would be better to update it for all steps with prefix

Reply:
fixed